src/NuGet.Clients/NuGet.CommandLine/DiagnosticCommands.cs

```diff
@@ -157,31 +158,49 @@ private int SummarizeLockfile(PackageSpec project, LockFile lockfile, string tar
             foreach (var target in targets)
             {
                 _log.LogMinimal($"Target: {target.TargetFramework} {target.RuntimeIdentifier}");
+                var sb = new StringBuilder();
                 foreach (var lib in target.Libraries)
                 {
-                    var provides = string.Empty;
+                    sb.Clear();
+
+                    sb.Append(" * [");
+
                     if (lib.NativeLibraries.Any())
                     {
-                        provides += Native + ",";
+                        sb.Append(Native);
+                        sb.Append(',');
                     }
                     if (lib.RuntimeAssemblies.Any())
                     {
-                        provides += Runtime + ",";
+                        sb.Append(Runtime);
+                        sb.Append(',');
                     }
                     if (lib.CompileTimeAssemblies.Any())
                     {
-                        provides += Compile + ",";
+                        sb.Append(Compile);
+                        sb.Append(',');
                     }
                     if (lib.FrameworkAssemblies.Any())
                     {
-                        provides += Framework + ",";
+                        sb.Append(Framework);
+                        sb.Append(',');
                     }
-                    provides = provides.TrimEnd(',');
-                    if (string.IsNullOrEmpty(provides))
+
+                    if (sb[sb.Length - 1] == ',')
                     {
-                        provides = Nothing;
+                        sb.Length--;
```

Comment:
This is not necessarily the same as the preceding code.  This code would only trim one trailing comma, whereas the previous code would trim all trailing commas.  This difference would be observable if values like `Compile`, `Framework`, etc. were null or a zero-length string.

Is that possible?